Source/WebCore/loader/PolicyChecker.cpp

```diff
@@ -177,6 +177,14 @@ void FrameLoader::PolicyChecker::checkNavigationPolicy(ResourceRequest&& request
     }
 #endif
 
+    auto originalRequest { request };
+#if ENABLE(WEB_ARCHIVE) && USE(CF)
+    constexpr auto webArchivePrefix = "webarchive+"_s;
+    auto url = request.url();
+    if (url.protocol().startsWith(webArchivePrefix))
```

Comment:
Is this really the best place to do this? What check was actually failing before your fix?
If it is a canHandleProtocol check, I am wondering if it wouldn't be better to strip `webarchive+` in that canHandleProtocol check, so that the fix is more localized.

Reply:
Ideally we should put this in the canHandleRequest check, but that wasn't sufficient when I tested it, so I believe this is the best place. If we delay stripping it, then we may never have the opportunity to strip that prefix before a navigation decision is made. For example, if a navigation delegate is defined and it has a navigation action decision handler, then WebKit won't check if it can handle the request.